### ISSUE TITLE
Configure nrepl file to watch for Jack-in

### ISSUE THREAD

**Issue by filipesilva** [BetterThanTomorrow/calva#254]
Heya, I use calva with a `shadow-cljs` project. When jacking-in to a build, two files with the nrepl port will be written: `.nrepl-port` and `.shadow-cljs/nrepl.port`. The first one doesn't necessarily contain an up to date port (see https://github.com/thheller/shadow-cljs/issues/524#issuecomment-510365911) and instead the second one should be used.

Is there a way to configure what file Calva should watch to determine the port? I tried looking over the Calva settings but I don't see anything obvious. 

The `calva.customCljsRepl` and `calva.jackInEnv` options seem promising though. If those are the right ones, is there an example of what these options can be set to somewhere? I tried the wiki but couldn't find anything.

**Comment by PEZ** [comment]
Hello,

I was looking at this today some. I think the way to go is to recognize that it is a shadow-cljs project and watch the right file. A complication is that there is also a **Connect to a running REPL** scenario which also looks for this file. I didn't figure out a good solution today. Maybe I should make it configurable in the interim...

**Comment by filipesilva** [comment]
Thanks for getting back to me! I'm in no rush really. The workaround is to delete the `.nrepl-port` file if it's there, and then `shadow-cljs` will generate a new one with the correct port. I can do that for now.

**Comment by PEZ** [comment]
Hello again. As a temporary measure I now made the Jack-in process start with deleting the file. Let me know if that helps you.

**Comment by filipesilva** [comment]
Just tested it and it works quite well, yes!

**Comment by PEZ** [comment]
Quite a lot has happened in this area. For the issue at hand, the most relevant change is that Calva now knows where to look for the port file with shadow-cljs. And then this port file is also configurable using [Customizable REPL Connect Sequences](https://github.com/BetterThanTomorrow/calva/wiki/Custom-Connect-Sequences).

**Comment by filipesilva** [comment]
Heya, thanks for letting me know about the new sequences. I had wondered if it was possible to not prompt for the jack in/cljs build and it looks like `menuSelections` allows me to do that, which is awesome!

**Comment by filipesilva** [comment]
I was going through the docs and saw that the `Calva built-in sequences` link didn't point anywhere, so I edited the wiki to point at https://github.com/BetterThanTomorrow/calva/blob/master/calva/nrepl/connectSequence.ts, which seems the right file.

Then I tried making a local workspace config for my shadow-cljs project ([repo here](https://github.com/filipesilva/learn-re-frame-course-files)). I wanted to auto-pick the `:app` build for both in initial build and the cljs repl. It ended up looking like this:
```
{
  "calva.replConnectSequences": [
    {
      "name": "custom-shadow-cljs",
      "projectType": "shadow-cljs",
      "cljsType": "shadow-cljs",
      "menuSelections": {
        "cljsLaunchBuilds": [":app"],
        "cljsDefaultBuild": ":app",
      }
    },
  ]
}
```

I expected `"cljsLaunchBuilds": [":app"],` to mean "always choose :app as the initial build" and `"cljsDefaultBuild": ":app",` to mean "always choose :app as the cljs repl". I looks like `"cljsDefaultBuild": ":app",` does what I expected, but I still have to pick the launch build manually by pressing enter. Is this expected?
